### ISSUE TITLE
kprove: Missing symbols when using Haskell backend

### ISSUE THREAD

**Issue by ttuegel** [runtimeverification/k#703]
`kprove` complains about missing symbols when compiling a KEVM proof for the Haskell backend:
```
$ ./kevm prove --backend haskell tests/specs/hkg-erc20/allowance-spec.k --debug --dry-run
java.util.NoSuchElementException: key not found: #symCmem(_)_EVM-SYMBOLIC__Int
	at scala.collection.MapLike.default(MapLike.scala:232)
	at scala.collection.MapLike.default$(MapLike.scala:231)
	at scala.collection.AbstractMap.default(Map.scala:59)
	at scala.collection.MapLike.apply(MapLike.scala:141)
	at scala.collection.MapLike.apply$(MapLike.scala:140)
	at scala.collection.AbstractMap.apply(Map.scala:59)
	at org.kframework.compile.AddSortInjections.production(AddSortInjections.java:305)
	at org.kframework.compile.AddSortInjections.sort(AddSortInjections.java:247)
	at org.kframework.compile.AddSortInjections.visitChildren(AddSortInjections.java:216)
	at org.kframework.compile.AddSortInjections.internalAddSortInjections(AddSortInjections.java:125)
	at org.kframework.compile.AddSortInjections.visitChildren(AddSortInjections.java:199)
	at org.kframework.compile.AddSortInjections.internalAddSortInjections(AddSortInjections.java:125)
	at org.kframework.compile.AddSortInjections.visitChildren(AddSortInjections.java:199)
	at org.kframework.compile.AddSortInjections.internalAddSortInjections(AddSortInjections.java:125)
	at org.kframework.compile.AddSortInjections.visitChildren(AddSortInjections.java:199)
	at org.kframework.compile.AddSortInjections.internalAddSortInjections(AddSortInjections.java:125)
	at org.kframework.compile.AddSortInjections.visitChildren(AddSortInjections.java:204)
	at org.kframework.compile.AddSortInjections.internalAddSortInjections(AddSortInjections.java:125)
	at org.kframework.compile.AddSortInjections.addTopSortInjections(AddSortInjections.java:116)
	at org.kframework.compile.AddSortInjections.addInjections(AddSortInjections.java:68)
	at org.kframework.backend.kore.ModuleToKORE.convertRule(ModuleToKORE.java:606)
	at org.kframework.backend.kore.ModuleToKORE.convertSpecificationModule(ModuleToKORE.java:594)
	at org.kframework.backend.haskell.HaskellRewriter$1.saveKoreSpecToTemp(HaskellRewriter.java:253)
	at org.kframework.backend.haskell.HaskellRewriter$1.prove(HaskellRewriter.java:306)
	at org.kframework.kprove.KProve.run(KProve.java:60)
	at org.kframework.kprove.KProveFrontEnd.run(KProveFrontEnd.java:92)
	at org.kframework.main.FrontEnd.main(FrontEnd.java:59)
	at org.kframework.main.Main.runApplication(Main.java:119)
	at org.kframework.main.Main.runApplication(Main.java:108)
	at org.kframework.main.Main.main(Main.java:54)
[Error] Internal: Uncaught exception thrown of type NoSuchElementException (key
not found: #symCmem(_)_EVM-SYMBOLIC__Int)
```

The symbol is defined in `evm-symbolic.k`, which is imported through `abstract-semantics-segmented-gas.k` in `allowance-spec.k`:
```
requires "abstract-semantics-segmented-gas.k"
requires "verification.k"

module ALLOWANCE-SPEC
  imports ETHEREUM-SIMULATION
  imports ABSTRACT-SEMANTICS-SEGMENTED-GAS
  imports VERIFICATION

...
```

**Comment by yzhang90** [comment]
We can import `ABSTRACT-SEMANTICS-SEGMENTED-GAS` in the `VERIFICATION` module, remove `imports ETHEREUM-SIMULATION` and pass `--def-module VERIFICATION` to ./kevm prove command

**Comment by denis-bogdanas** [comment]
Is this still an issue?

**Comment by ehildenb** [comment]
This will be solved by not allowing new syntax between the `--def-module` and the `--spec-module`, which is addressed in #1681
